lib/Doctrine/DBAL/Driver/SQLAnywhere/SQLAnywhereStatement.php

```diff
@@ -73,7 +73,7 @@ class SQLAnywhereStatement implements IteratorAggregate, Statement
     public function __construct($conn, $sql)
     {
         if (! is_resource($conn)) {
-            throw new SQLAnywhereException('Invalid SQL Anywhere connection resource: ' . $conn);
+            throw new SQLAnywhereException(sprintf('Invalid SQL Anywhere connection resource "%s".', $conn));
```

Comment:
This won't be informative if the value is not a scalar (and it's not expected to be a scalar). Likely, it won't happen often since we enforce types statically, but otherwise, we need a simple formatting function which can indicate (not dump) an arbitrary value (e.g. class if it's an object, the value itself if it's a string or number, etc).